broker/cloud_run/lsst/classify_snn/main.py

```diff
@@ -0,0 +1,170 @@
+        tuple into a proper HTTP response. Note that the response is a status message for the web server
+        and should not contain the classification results.
+    """
+    # extract the envelope from the request that triggered the endpoint
+    # this contains a single Pub/Sub message with the alert to be processed
+    envelope = flask.request.get_json()
+
+    # unpack the alert. raises a `BadRequest` if the envelope does not contain a valid message
+    try:
+        alert = pittgoogle.Alert.from_cloud_run(envelope, "lsst")
+    except pittgoogle.exceptions.BadRequest as exc:
+        return str(exc), HTTP_400
+
+    # classify
+    snn_dict = _classify(alert)
+
+    # prepare data for publishing
+    classifier_summary = _classification_summary(snn_dict)
+    snn_alert = _create_outgoing_alert(alert, snn_dict)
+    snn_results = pittgoogle.Alert.from_dict(payload=snn_dict)
+
+    # publish
+    TOPIC.publish(snn_alert, serializer="avro")
+    TOPIC_BIGQUERY_IMPORT_CLASSIFICATIONS.publish(classifier_summary, serializer="json")
+    TOPIC_BIGQUERY_IMPORT_SUPERNNOVA.publish(snn_results, serializer="json")
+
+    return "", HTTP_204
+
+
+def _classify(alert: pittgoogle.Alert) -> dict:
+    """Classify the alert using SuperNNova."""
+    # init
+    snn_df = _format_for_classifier(alert)
+    device = "cpu"
+
+    # classify
+    _, pred_probs = classify_lcs(snn_df, MODEL_PATH, device)
+
+    # use `.item()` to convert numpy -> python types for later serialization
+    pred_probs = pred_probs.flatten()
+    snn_dict = {
+        "diaObjectId": alert.objectid,
+        "diaSourceId": alert.sourceid,
+        "prob_class0": pred_probs[0].item(),
+        "prob_class1": pred_probs[1].item(),
+        "predicted_class": np.argmax(pred_probs).item(),
+    }
+
+    return snn_dict
+
+
+def _format_for_classifier(alert: pittgoogle.Alert) -> pd.DataFrame:
+    """Create a DataFrame for input to SuperNNova."""
+    alert_df = alert.dataframe
+    snn_df = pd.DataFrame(
+        data={
+            # select a subset of columns and rename them for SuperNNova
+            # get_key returns the name that the survey uses for a given field
+            # for the full mapping, see alert.schema.map
+            "SNID": [alert.objectid] * len(alert_df.index),
+            "FLT": alert_df[alert.get_key("filter")[1]],
+            "MJD": alert_df[alert.get_key("mjd")[1]],
+            "FLUXCAL": alert_df[alert.get_key("flux")[1]],
+            "FLUXCALERR": alert_df[alert.get_key("flux_err")[1]],
+        },
+        index=alert_df.index,
+    )
+
+    return snn_df
+
+
+def _create_outgoing_alert(alert: pittgoogle.Alert, snn_dict: dict) -> pittgoogle.Alert:
+    """Combine the incoming alert with the classification results to create the outgoing alert."""
+
+    return pittgoogle.Alert.from_dict(
+        payload={**alert.dict, "SuperNNova": snn_dict},
+        attributes={"supernnova_class": snn_dict["predicted_class"], **alert.attributes},
+    )
```

Comment:
If we're going with a "lite" structure that looks like this:

```
'alert': {...}
'value-added': {our stuff...}
```

then let's do the same thing here. But tbh, I'm not totally sold on that because I think it's going to complicate a few things for us. One is how the `Alert` methods handle the data. All of a sudden there's this extra layer that it's not expecting.

Either way, I think `pittgoogle.Alert` should have a method that accepts a value-added dict and adds it to the payload so that the user doesn't have to do this manually. But let's sort out what we want it to look like first.